src/export/InstanceExporter.ts

```diff
@@ -447,6 +455,74 @@ export class InstanceExporter implements Fishable {
+    fshDef: Instance,
+    instanceDef: InstanceDefinition,
+    instanceOfStructureDefinition: StructureDefinition
+  ) {
+    const nonChoiceSlicedElements = instanceOfStructureDefinition.elements.filter(
+      e => e.slicing && !e.path.endsWith('[x]')
+    );
+    for (const slicingEl of nonChoiceSlicedElements) {
+      // Use the sliced element's path to get all potential element values from the instance
+      const values = collectValuesAtElementIdOrPath(slicingEl.id, instanceDef);
+      // Find the values that do not have named slices
+      const unnamedValues = values.values.filter(v => v != null && v._sliceName == null);
+      if (unnamedValues.length) {
+        // Find the values that do have named slices (for comparison)
+        const namedValues = values.values
+          .filter(v => v?._sliceName != null)
+          .map(v => {
+            const cleanValue = cloneDeep(v);
+            delete cleanValue._sliceName;
+            return { name: v._sliceName, value: cleanValue };
+          });
+        if (namedValues.length) {
+          for (const value of unnamedValues) {
+            // It's a match if the unnamed value is a compatible superset of the named value (i.e., slice)
+            const matches = namedValues
+              .filter(nv => matchesPattern(value, nv.value))
+              .map(nv => nv.name);
+            if (matches.length) {
+              if (namedValues.every(nm => isEqual(value, nm.value))) {
+                logger.warn(
+                  `${instanceDef.id} has an array item that is exactly the same as a required slice, but does not ` +
+                    'use the slice name in its path. Usually you can remove this item from your FSH since FSH ' +
+                    'instances inherit required assigned values from their profile. Alternately, you can modify ' +
+                    "your FSH to use the slice name in the item's path. See: " +
+                    'https://hl7.org/fhir/uv/shorthand/reference.html#sliced-array-paths\n' +
+                    `  Path:  ${slicingEl.path}\n` +
+                    `  Slice: ${matches.join(' or ')}\n` +
+                    `  Value: ${JSON.stringify(value)}`,
+                  fshDef.sourceInfo
+                );
+              } else {
+                logger.warn(
+                  `${instanceDef.id} has an array item that matches a required slice, but does not use the slice name ` +
+                    'in its path. This may result in unexpected items in your array. If the item is intended to be ' +
+                    "in the slice, modify your FSH to use the slice name in the item's path. See: " +
+                    'https://hl7.org/fhir/uv/shorthand/reference.html#sliced-array-paths\n' +
+                    `  Path:  ${slicingEl.path}\n` +
+                    `  Slice: ${matches.join(' or ')}\n` +
+                    `  Value: ${JSON.stringify(value)}`,
+                  fshDef.sourceInfo
+                );
```

Comment:
Thinking about these two messages together, I think it would be good if the first message also included the part about unexpected items. So maybe the first one could be something like:
> `${instanceDef.id}` has an array item that is exactly the same as a required slice, but does not use the slice name in its path. This may result in unexpected items in your array. Usually, you can remove this item from your FSH, since FSH instances inherit required assigned values from their profile. Alternately, you can modify your FSH to use the slice name in the item's path.

Reply:
Addressed in [4fea2ef](https://github.com/FHIR/sushi/pull/1200/commits/4fea2efe5f1365762afddea29f0838f921ec72b0).